### ISSUE TITLE
Allow selective disabling of autocomplete lists

### ISSUE THREAD

**Issue by nathanlesage** [Zettlr/Zettlr#3092]
### Description

While autocomplete is a very handy addition to Zettlr and speeds up many processes, for some workflows or in some instances it can be very unwieldy. I have already attempted to make it as unobtrusive as possible (i.e. the autocomplete dropdown disappears very fast and never overwrites anything automatically), but, as it turns out, for some workflows, even that could be too much — especially if you never use a feature at all.

### Proposed Changes

Add settings toggles to programmatically disable certain autocomplete lists. We can basically implement a check for these settings inside the `shouldAutocomplete` hook in the Markdown editor. The changes should be straight forward.

### Caveats

As long as we keep the autocompletes on by default, nothing should change for people. The only thing we have to remember is to access the options from the cm instance, and not pull them in from main via `global.config.get`.

### Do you Wish to Attempt Implementing this Yourself?

Yes

### Zettlr Version

Right off the develop branch

### Your Platform

- [X] Windows
- [X] macOS
- [X] Linux

### Operating System Version

Platform-independent

### Additional Information

See issue #3082

**Comment by ailintom** [comment]
I would also like to add that it would be nice to configure sort order in autocomplete dropdown lists or to disable autocomplete by pressing ENTER, leaving only TAB. Items are now sorted by most commonly used and since dropdowns autocomplete automatically upon pressing ENTER as well as upon pressing TAB, you never get what you want. 
If you just type ` ```r ` and press ENTER , instead of a code block formatted in R you are getting a code block in formatted for another programming language (Javascript in Zettlr 2.3.0 or Clojure in Zettlr 3.0.0-beta5-nightly)

**Comment by jtherrmann** [comment]
I came here from https://github.com/Zettlr/Zettlr/issues/3082 and would also love the option to disable the tag autocomplete popup after typing `#`, as I don't use tags and the autocomplete popup is distracting when creating headings.

**Comment by exanrauzer** [comment]
> I would also like to add that it would be nice to configure sort order in autocomplete dropdown lists or to disable autocomplete by pressing ENTER, leaving only TAB. Items are now sorted by most commonly used and since dropdowns autocomplete automatically upon pressing ENTER as well as upon pressing TAB, you never get what you want. If you just type `` ```r `` and press ENTER , instead of a code block formatted in R you are getting a code block in formatted for another programming language (Javascript in Zettlr 2.3.0 or Clojure in Zettlr 3.0.0-beta5-nightly) This would not happen if the list of programming languages was sorted alphabetically or is you had to press TAB for autocomplete

Yes, items in the code list should be sorted according to the signs you typed and alphabetically. However, I would opt to maintain both the tab and enter workflow, as it may vary for each user.

**Comment by exanrauzer** [comment]
> I came here from #3082 and would also love the option to disable the tag autocomplete popup after typing `#`, as I don't use tags and the autocomplete popup is distracting when creating headings.

Nice catch.
